vendor/github.com/docker/distribution/registry/handlers/images.go

```diff
@@ -95,7 +95,13 @@ func (imh *imageManifestHandler) GetImageManifest(w http.ResponseWriter, r *http
 	}
 	manifest, err = manifests.Get(imh, imh.Digest, options...)
 	if err != nil {
-		imh.Errors = append(imh.Errors, v2.ErrorCodeManifestUnknown.WithDetail(err))
+		switch err.(type) {
+		case errcode.ErrorCode, errcode.Error:
+			// keep known error as is
+		default:
+			err = v2.ErrorCodeManifestUnknown.WithDetail(err)
+		}
+		imh.Errors = append(imh.Errors, err)
```

Comment:
we need to do this as a carry commit to our fork of the docker distribution repo and then bump it here, right?

Reply:
(and probably add a test to make sure we don't accidentally drop the commit at some point)